nanoPU_sim.py

```diff
@@ -478,13 +458,11 @@ def timeoutEvent(self, tx_msg_id, rtx_offset):
             rtx_pkts_mask = (1<<rtx_offset)-1
             rtx_pkts = ~delivered_bitmap & rtx_pkts_mask
             self.log('Pkts to retransmit: {:b}'.format(rtx_pkts))
-            toBtx_bitmap = self.toBtx[tx_msg_id]
-            self.toBtx[tx_msg_id] = toBtx_bitmap | rtx_pkts
             # reschedule timer for this msg
             self.rescheduleTimerEvent(tx_msg_id, self.max_tx_pkt_offset[tx_msg_id])
             if rtx_pkts != 0:
-                # make the message active
-                self.enq_active_messages_fifo(tx_msg_id)
+                # schedule the pkts for transmission
+                self.scheduled_pkts_fifo.put((tx_msg_id, rtx_pkts))
```

Comment:
Does this mean the scheduled_pkts_fifo can have two entries that belong to the same msg (one inserted by the creditToBTx event and one inserted by the Timeout event)

Reply:
Yes, that's right. Both the timeout event and the creditToBtx event can schedule pkts for transmission. They should be scheduling different pkts within the msg.

Reply:
awesome! Thanks for clarification.